src/verify_expr.ml

```diff
@@ -1841,30 +1858,41 @@ module VerifyExpr(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
       let rec iter chunks vs h fields =
         match fields with
           [] ->
-          let (_, csym, _, _) = List.assoc sn struct_accessor_map in
-          cont chunks h (Some (if consumeUninitChunk then real_unit (* dummy term; should never be used *) else ctxt#mk_app csym (List.rev vs)))
+          let (_, csym, _, _, csym_opt) = List.assoc sn struct_accessor_map in
+          let csym_symb = if consumeUninitChunk then csym_opt else csym in
+          let struct_val = ctxt#mk_app csym_symb (List.rev vs) in
+          cont chunks h (Some struct_val)
         | (f, (lf, gh, t0, offset, finit))::fields ->
           let t = instantiate_type tpenv t0 in
+          let prover_convert_term_as_option_types term =
+            prover_convert_term term (option_type t) (option_type t0) 
+          in

```

Comment:
This is pointless: all option types are of ProvertypeInductive, so no conversion required.